edge-hub/test/Microsoft.Azure.Devices.Routing.Core.Test/RouteTest.cs

```diff
@@ -19,29 +20,29 @@ public class RouteTest : RoutingUnitTestBase
         static readonly Endpoint Endpoint2 = new TestEndpoint("id2");
         static readonly Endpoint Endpoint3 = new TestEndpoint("id1");
 
-        static readonly Route Route1 = new Route("id1", "rule1", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint> { Endpoint1 });
-        static readonly Route Route2 = new Route("id1", "rule1", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint> { Endpoint2 });
-        static readonly Route Route3 = new Route("id1", "rule1", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint> { Endpoint3 });
-        static readonly Route Route4 = new Route("id2", "rule2", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint> { Endpoint1 });
-        static readonly Route Route5 = new Route("id3", "rule3", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint>());
-        static readonly Route Route6 = new Route("id3", "rule3", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint>());
-        static readonly Route Route7 = new Route("id2", "rule1", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint> { Endpoint1 });
+        static readonly Route Route1 = new Route("id1", "rule1", "hub", TelemetryMessageSource.Instance, Endpoint1, 0, 0);
+        static readonly Route Route2 = new Route("id1", "rule1", "hub", TelemetryMessageSource.Instance, Endpoint2, 0, 0);
+        static readonly Route Route3 = new Route("id1", "rule1", "hub", TelemetryMessageSource.Instance, Endpoint3, 0, 0);
+        static readonly Route Route4 = new Route("id2", "rule2", "hub", TelemetryMessageSource.Instance, Endpoint1, 0, 0);
+        static readonly Route Route5 = new Route("id3", "rule3", "hub", TelemetryMessageSource.Instance, new Mock<Endpoint>("id1").Object, 0, 0);
+        static readonly Route Route6 = new Route("id3", "rule3", "hub", TelemetryMessageSource.Instance, new Mock<Endpoint>("id1").Object, 0, 0);
+        static readonly Route Route7 = new Route("id2", "rule1", "hub", TelemetryMessageSource.Instance, Endpoint1, 0, 0);
 
         [Fact]
         [Unit]
         public void TestConstructor()
         {
-            Assert.Throws<ArgumentNullException>(() => new Route(null, "condition", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint>()));
-            Assert.Throws<ArgumentNullException>(() => new Route("id", null, "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint>()));
-            Assert.Throws<ArgumentNullException>(() => new Route("id", "condition", null, TelemetryMessageSource.Instance, new HashSet<Endpoint>()));
-            Assert.Throws<ArgumentNullException>(() => new Route("id", "condition", "hub", TelemetryMessageSource.Instance, null));
+            Assert.Throws<ArgumentNullException>(() => new Route(null, "condition", "hub", TelemetryMessageSource.Instance, new Mock<Endpoint>("id1").Object, 0, 0));
```

Comment:
nit: try to use [Theory] and [InlineData]

Reply:
out of scope here, trying to keep changes to existing test collateral to a minimum (or I won't ever finish).

---
In reply to: [377462258](https://github.com/Azure/iotedge/pull/2469#discussion_r377462258) [](ancestors = 377462258)